options.go

```diff
@@ -45,6 +45,9 @@ type CloneOptions struct {
 	ReferenceName plumbing.ReferenceName
 	// Fetch only ReferenceName if true.
 	SingleBranch bool
+	// Mirror clones the repository as a mirror.
+	// This implies bare and fetches all refs.
+	Mirror bool
```

Comment:
We could expand a bit more on the documentation to align with upstream.

```
Compared to --bare, --mirror not only maps local branches of the source to local branches of the target, it maps all refs (including remote-tracking branches, notes etc.) and sets up a refspec configuration such that all these refs are overwritten by a git remote update in the target repository.
```
https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---mirror

Reply:
Done